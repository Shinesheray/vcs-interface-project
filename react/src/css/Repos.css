.repo-block {
  margin-top: 20px;
}

.repo-heading {
  background-color: rgba(31, 160, 224);
  border-radius: 10px;
  color: white;
  height: fit-content;
  margin-bottom: 0px;
  padding-bottom: 0px;
}

.repo-heading p {
  padding: 10px;
  margin: 0px;
  color: white;
  font-weight: bold;
}

.date-created-div p {
  color: rgb(31, 160, 224);
  font-size: 10px;
  text-align: right;
  margin: 0px;
  padding: 2px 10px;
}

.repo-commit-heading {
  background-color: rgb(255, 255, 255);
  height: fit-content;
}
.repo-commit-heading p {
  padding: 5px 0px 5px 10px;
  margin: 0px;
  font-size: 80%;
  color: black;
  font-style: italic;
}

.repo-desc {
  color: white;
  font-size: 14px;
  font-weight: normal !important;
}

.repo-commits {
  padding: 10px;
}
.repo-commits p {
  margin: 0px;
  padding: 0px;
  color: rgb(31, 160, 224);
  font-size: 15px;
}

ul {
  margin-left: 10px;
}
