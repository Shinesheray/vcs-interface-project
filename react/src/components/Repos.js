import React from "react";
import RepoCommits from "./RepoCommits";
import "../css/Repos.css";

export default function Repos(props) {
  const { repos } = props;

  // Map repos
  let repo = repos.map((item, index) => {
    return (
      <div key={index}>
        <div className="repo-block">
          {/* Repo details Section */}
          <div className="repo-heading">
            {/* Show "No Description" if none */}
            <p>
              {item.repoName} <br></br>{" "}
              <span className="repo-desc">
                {item.repoDesc === undefined ? "No Description" : item.repoDesc}
              </span>
            </p>
            <div className="date-created-div">
              <p>Created - {item.createdOn}</p>
            </div>
          </div>
          {/* Commit section */}
          <div className="repo-commit-heading">
            <p>Commit History</p>
          </div>
          {/* Details div, to the right of the image */}
          <div className="repo-commits">
            <RepoCommits commitMsg={item.commitMsg} />
          </div>
        </div>
      </div>
    );
  });

  return <div>{repo}</div>;
}
