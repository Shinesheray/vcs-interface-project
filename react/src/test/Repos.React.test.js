import React from "react";
import renderer from "react-test-renderer";
import Repos from "../Repos";

// Empty set of results
const reposEmptyExample = [];
// Full set of results
const reposFullExample = [
  {
    repoName: "repo1",
    repoDesc: "repo1Desc",
    createdOn: "dd-mm-yy",
    commitMsg: ["msg1", "msg2", "msg3", "msg4", "msg5"],
  },
  {
    repoName: "repo2",
    repoDesc: "repo2Desc",
    createdOn: "dd-mm-yy",
    commitMsg: ["msg1", "msg2", "msg3", "msg4", "msg5"],
  },
  {
    repoName: "repo3",
    repoDesc: "repo3Desc",
    createdOn: "dd-mm-yy",
    commitMsg: ["msg1", "msg2", "msg3", "msg4", "msg5"],
  },
  {
    repoName: "repo4",
    repoDesc: "repo4Desc",
    createdOn: "dd-mm-yy",
    commitMsg: ["msg1", "msg2", "msg3", "msg4", "msg5"],
  },
  {
    repoName: "repo5",
    repoDesc: "repo5Desc",
    createdOn: "dd-mm-yy",
    commitMsg: ["msg1", "msg2", "msg3", "msg4", "msg5"],
  },
];
// Partial set of results ( this is mainly for bitbucket )
const reposPartialExample = [
  {
    repoName: "repo1",
    repoDesc: "repo1Desc",
    createdOn: "dd-mm-yy",
    commitMsg: ["msg1", "msg2", "msg3", "msg4", "msg5"],
  },
  {
    repoName: "repo2",
    createdOn: "dd-mm-yy",
    commitMsg: ["msg1", "msg4", "msg5"],
  },
  { repoName: "repo3", repoDesc: "", createdOn: "dd-mm-yy", commitMsg: [] },
];

// Test case for a full repo with all details
test("Renders full repos correctly", () => {
  const tree = renderer.create(<Repos repos={reposFullExample} />).toJSON();

  expect(tree).toMatchSnapshot();
});

// Test case for a partial repo with some details
test("Renders partial repos correctly", () => {
  const tree = renderer.create(<Repos repos={reposPartialExample} />).toJSON();

  expect(tree).toMatchSnapshot();
});

// Test case for a empty repo
test("Renders empty repos correctly", () => {
  const tree = renderer.create(<Repos repos={reposEmptyExample} />).toJSON();

  expect(tree).toMatchSnapshot();
});
