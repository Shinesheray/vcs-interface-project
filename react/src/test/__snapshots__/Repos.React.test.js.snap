// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders empty repos correctly 1`] = `<div />`;

exports[`Renders full repos correctly 1`] = `
<div>
  <div>
    <div
      className="repo-block"
    >
      <div
        className="repo-heading"
      >
        <p>
          repo1
           
          <br />
           
          <span
            className="repo-desc"
          >
            repo1Desc
          </span>
        </p>
        <div
          className="date-created-div"
        >
          <p>
            Created - 
            dd-mm-yy
          </p>
        </div>
      </div>
      <div
        className="repo-commit-heading"
      >
        <p>
          Commit History
        </p>
      </div>
      <div
        className="repo-commits"
      >
        <ul>
          <li>
            msg1
          </li>
          <li>
            msg2
          </li>
          <li>
            msg3
          </li>
          <li>
            msg4
          </li>
          <li>
            msg5
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div>
    <div
      className="repo-block"
    >
      <div
        className="repo-heading"
      >
        <p>
          repo2
           
          <br />
           
          <span
            className="repo-desc"
          >
            repo2Desc
          </span>
        </p>
        <div
          className="date-created-div"
        >
          <p>
            Created - 
            dd-mm-yy
          </p>
        </div>
      </div>
      <div
        className="repo-commit-heading"
      >
        <p>
          Commit History
        </p>
      </div>
      <div
        className="repo-commits"
      >
        <ul>
          <li>
            msg1
          </li>
          <li>
            msg2
          </li>
          <li>
            msg3
          </li>
          <li>
            msg4
          </li>
          <li>
            msg5
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div>
    <div
      className="repo-block"
    >
      <div
        className="repo-heading"
      >
        <p>
          repo3
           
          <br />
           
          <span
            className="repo-desc"
          >
            repo3Desc
          </span>
        </p>
        <div
          className="date-created-div"
        >
          <p>
            Created - 
            dd-mm-yy
          </p>
        </div>
      </div>
      <div
        className="repo-commit-heading"
      >
        <p>
          Commit History
        </p>
      </div>
      <div
        className="repo-commits"
      >
        <ul>
          <li>
            msg1
          </li>
          <li>
            msg2
          </li>
          <li>
            msg3
          </li>
          <li>
            msg4
          </li>
          <li>
            msg5
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div>
    <div
      className="repo-block"
    >
      <div
        className="repo-heading"
      >
        <p>
          repo4
           
          <br />
           
          <span
            className="repo-desc"
          >
            repo4Desc
          </span>
        </p>
        <div
          className="date-created-div"
        >
          <p>
            Created - 
            dd-mm-yy
          </p>
        </div>
      </div>
      <div
        className="repo-commit-heading"
      >
        <p>
          Commit History
        </p>
      </div>
      <div
        className="repo-commits"
      >
        <ul>
          <li>
            msg1
          </li>
          <li>
            msg2
          </li>
          <li>
            msg3
          </li>
          <li>
            msg4
          </li>
          <li>
            msg5
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div>
    <div
      className="repo-block"
    >
      <div
        className="repo-heading"
      >
        <p>
          repo5
           
          <br />
           
          <span
            className="repo-desc"
          >
            repo5Desc
          </span>
        </p>
        <div
          className="date-created-div"
        >
          <p>
            Created - 
            dd-mm-yy
          </p>
        </div>
      </div>
      <div
        className="repo-commit-heading"
      >
        <p>
          Commit History
        </p>
      </div>
      <div
        className="repo-commits"
      >
        <ul>
          <li>
            msg1
          </li>
          <li>
            msg2
          </li>
          <li>
            msg3
          </li>
          <li>
            msg4
          </li>
          <li>
            msg5
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`Renders partial repos correctly 1`] = `
<div>
  <div>
    <div
      className="repo-block"
    >
      <div
        className="repo-heading"
      >
        <p>
          repo1
           
          <br />
           
          <span
            className="repo-desc"
          >
            repo1Desc
          </span>
        </p>
        <div
          className="date-created-div"
        >
          <p>
            Created - 
            dd-mm-yy
          </p>
        </div>
      </div>
      <div
        className="repo-commit-heading"
      >
        <p>
          Commit History
        </p>
      </div>
      <div
        className="repo-commits"
      >
        <ul>
          <li>
            msg1
          </li>
          <li>
            msg2
          </li>
          <li>
            msg3
          </li>
          <li>
            msg4
          </li>
          <li>
            msg5
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div>
    <div
      className="repo-block"
    >
      <div
        className="repo-heading"
      >
        <p>
          repo2
           
          <br />
           
          <span
            className="repo-desc"
          >
            No Description
          </span>
        </p>
        <div
          className="date-created-div"
        >
          <p>
            Created - 
            dd-mm-yy
          </p>
        </div>
      </div>
      <div
        className="repo-commit-heading"
      >
        <p>
          Commit History
        </p>
      </div>
      <div
        className="repo-commits"
      >
        <ul>
          <li>
            msg1
          </li>
          <li>
            msg4
          </li>
          <li>
            msg5
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div>
    <div
      className="repo-block"
    >
      <div
        className="repo-heading"
      >
        <p>
          repo3
           
          <br />
           
          <span
            className="repo-desc"
          >
            
          </span>
        </p>
        <div
          className="date-created-div"
        >
          <p>
            Created - 
            dd-mm-yy
          </p>
        </div>
      </div>
      <div
        className="repo-commit-heading"
      >
        <p>
          Commit History
        </p>
      </div>
      <div
        className="repo-commits"
      >
        <ul />
      </div>
    </div>
  </div>
</div>
`;
